device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/TopicParser.java

```diff
@@ -36,25 +36,23 @@ protected String getStatus(int tokenIndexStatus) throws TransportException
             throw new TransportException(new IllegalArgumentException("Invalid token Index for status"));
         }
 
-        if (topicTokens.length > tokenIndexStatus)
-        {
-            String token = topicTokens[tokenIndexStatus];
+        String token = topicTokens[tokenIndexStatus];
 
-            if (token != null)
-            {
-                //Codes_SRS_TopicParser_25_004: [This method shall return the status corresponding to the tokenIndexStatus from tokens if it is not null.]
-                status = token;
-            }
-            else
-            {
-                //Codes_SRS_TopicParser_25_005: [If token corresponding to tokenIndexStatus is null then this method shall throw TransportException.]
-                throw new TransportException("Status could not be parsed");
-            }
+        if (token != null)
+        {
+            //Codes_SRS_TopicParser_25_004: [This method shall return the status corresponding to the tokenIndexStatus from tokens if it is not null.]
+            status = token;
+        }
+        else
+        {
+            //Codes_SRS_TopicParser_25_005: [If token corresponding to tokenIndexStatus is null then this method shall throw TransportException.]
+            throw new TransportException("Status could not be parsed");
         }
 
         return status;
     }
 
+    @SuppressWarnings("SameParameterValue") // Method is designed to be generic.
```

Comment:
what's the warning here?

Reply:
this method currently has a single caller, and hence tokenIndexReqID currently has a single value